### PR TITLE
state management with getX

### PR DESCRIPTION
**Pull request by kk-boss**
- Separation of business logic from widgets with GetX

- All widgets are made stateless.

- Better state management.  Only parts of widgets that needs to rebuild will rebuild.

### CODE REVIEW

File: lib/Utils/methods.dart [hunk 1]
```diff
@@ -43,6 +45,14 @@ Future cropImage(String imagepath, Function(String) onCrop) async {
 
 void exportPdf(List<File> images) {
   final Document pdf = Document();
+  final fileToImage = images
```
Comment (cimplesid) [inline lib/Utils/methods.dart:48]: typo  ```filesToImage```

File: lib/controllers/images_to_pdf.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import 'dart:io';
+
```
Comment (cimplesid) [inline lib/controllers/images_to_pdf.dart:2]: we are following this single group import so please do that for all files.Better user bhyrang extension in vscode and put em in ascending just for the shake of sanity

File: lib/screens/images_to_pdf.dart [hunk 3]
```diff
@@ -1,63 +1,37 @@
...
 
-  void addImage(String path) => setState(() {
-        _images.add(File(path));
-      });
-  bool _cropImage = false;
+class ImageToPdf extends StatelessWidget {
+  final imageToPdfController = Get.put(ImageToPdfController());
   @override
   Widget build(BuildContext context) {
+    final images = imageToPdfController.images;
```
Comment (cimplesid) [inline lib/screens/images_to_pdf.dart:11]: ```_images```  sanity again

File: lib/screens/scan_and_convert.dart [hunk 4]
```diff
@@ -1,63 +1,35 @@
...
 
+import 'package:EasyScan/controllers/scan_and_convert.dart';
 import 'package:EasyScan/Utils/constants.dart';
-import 'package:EasyScan/Utils/methods.dart';
-
-import 'package:flutter/material.dart';
 
-import 'package:image_picker/image_picker.dart';
+class ScanAndConvert extends StatelessWidget {
+  final scanAndConvertController = Get.put(ScanAndConvertController());
```
Comment (cimplesid) [inline lib/screens/scan_and_convert.dart:8]: ```_scanAndConvertController ```

File: lib/screens/scan_and_convert.dart [hunk 5]
```diff
@@ -1,63 +1,35 @@
...
-    } else {
-      _hasNotPickedImage = true;
+      return Image.file(scanAndConvertController.imageFile);
     }
-    setState(() {});
   }
 
   @override
   Widget build(BuildContext context) {
+    scanAndConvertController.getImageFromSource();
```
Comment (cimplesid) [inline lib/screens/scan_and_convert.dart:32]: since we don't have tests in this project.So can you just verify this don't cause stackoverflow

**Review by cimplesid — CHANGES_REQUESTED** [review]

**Comment by cimplesid** [comment]
@kk-boss  i have highlighted the some of them but make sure to use _ in the class level private variables.Else LGTM

File: lib/screens/scan_and_convert.dart (hunk 5 above)
Comment (kk-boss) [reply lib/screens/scan_and_convert.dart]: I have verified it, only the body listens to changes and build method runs just once. Though, I tweaked it in latest commit to remove confusions.

File: lib/controllers/images_to_pdf.dart (hunk 2 above)
Comment (kk-boss) [reply lib/controllers/images_to_pdf.dart:2]: I couldn't find the particular extension you mentioned. I have organized imports in as

Comment (cimplesid) [reply lib/controllers/images_to_pdf.dart:2]: search using this ```publisher:"pratishshr"``` in vscode

**Comment by cimplesid** [comment]
@kk-boss  are you sure you have pushed the commit resolving the imports ascending

File: lib/controllers/images_to_pdf.dart (hunk 2 above)
Comment (kk-boss) [reply lib/controllers/images_to_pdf.dart:2]: image

Looks like I couldn't get it right. I arranged them in alphabetical order.

Is the ladder format is what you want?

Comment (cimplesid) [reply lib/controllers/images_to_pdf.dart:2]: yes ascending by length,that's ladder

Comment (kk-boss) [reply lib/controllers/images_to_pdf.dart]: Resolved in latest commit 👍

**Review by cimplesid — APPROVED** [review]

**Pull request merged**